plugin/pkg/scheduler/factory/factory.go

```diff
@@ -223,6 +223,13 @@ func NewConfigFactory(
 					if err := c.podQueue.Delete(obj.(*v1.Pod)); err != nil {
 						runtime.HandleError(fmt.Errorf("unable to dequeue %T: %v", obj, err))
 					}
+					pod, ok := obj.(*v1.Pod)
+					if !ok {
+						// This shouldn't happen
+						return
+					}
+					// Volume binder only wants to keep unassigned pods
+					c.volumeBinder.DeletePodBindings(pod)
```

Comment:
@bsalamat can you verify that these are all the right places to clean up my cache?   I want to clean it up anytime the Pod is no longer unassigned.

And also the scenario where the pod gets deleted/updated while being processed in scheduleOne().

Reply:
This looks correct to me. This delete will be triggered when:
1. An unassigned pod is scheduled and bound.
2. An unassigned pod is actually deleted.

I am just not so sure about a case where an unassigned pod is updated and its volume claims change. Do we allow such an update at all? If we do, I guess we need invalidating the cache.

Reply:
Thanks.  You cannot modify pod.Spec.Volumes.